src/cobalt.rs

```diff
@@ -154,11 +172,42 @@ fn generate_pages(posts: Vec<Document>, documents: Vec<Document>, context: &Cont
         .collect();
 
     trace!("Generating other documents");
-    for mut doc in documents {
-        trace!("Generating {}", doc.url_path);
-        generate_doc(&posts_data, &mut doc, context)?;
+    for mut doc in pages {
+        trace!("Generating {} / {:?}", doc.url_path, doc.file_path.to_str());
+        let mut pagination_cfg = pagination_config::PaginationCfg::new(&doc.front.pagination);
+        if pagination_cfg.is_pagination_enable() {
+            trace!("It's an index page {}", doc.url_path);
+            let paginators =
+                pagination::generate_paginators(&mut doc, &posts_data, &mut pagination_cfg);
+            // page 1 is not in "_p" folder
```

Comment:
Ok, so we're baking in the difference between the permalinks?  This was something I brought up earlier but I thought you were planning to leave it alone.  I'm not criticizing, just trying to better understand.

Could you elaborate on how you expect this to work?

Reply:
For the `all`, it works like this: I'm creating a `paginator`  per page filling them with correct `paqe` number and links to next/previous page.

For the `tags` it does not work now that I see better what we need, so this will be reworked.

Reply:
> > Could you elaborate on how you expect this to work?

I'm not as much concerned about the implementation atm but the policy.

Reply:
Policy wise, I'm intending to generate all the different paginator (replacing the `{{ num }}` basically) in this function and use `generate_doc` to create the file, modifying `file_path` according to the `pagination.permalink`.

I don't know if I'm answering properly to your question here ^^'

Reply:
That is still implementation and not policy.

From looking at the code, the answer is that
- The "first" page is `front.permalink`
- All other pages use `front.pagination.permalink`

One problem with this code is that the comment does not say this.  Instead the comment makes it sound like somewhere else we have a baked-in policy.